src-v2/Judge/NTS.Judge/Events/CompetitionChildrenBehind.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Not.Application.Ports.CRUD;
+using Not.Blazor.Ports.Behinds;
+using NTS.Domain.Setup.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NTS.Judge.Events;
+public class CompetitionChildrenBehind : INotBehindParent<Contestant>
+{
+    private readonly IRepository<Competition> _competitionRepository;
+    private readonly IParentRepository<Contestant> _contestantRepository;
+    private Competition _competition;
+
+    public CompetitionChildrenBehind(IParentRepository<Contestant> contestants)
```

Comment:
The following feels like it must be conceptually wrong but I seem to lack the knowledge to reason why.
 ```C#
   public CompetitionChildrenBehind(IRepository<Competition> competition, IParentRepository<Contestant> contestants, Competition competitionParent)
   {
       _competitionRepository = competition;
       _contestantRepository = contestants;
   }
```